recursion_exercise/recursion.js

```diff
@@ -0,0 +1,56 @@
+function productOfArray(arr) {
+    if(arr.length === 1) return arr[0]
+    return arr[0] * productOfArray(arr.slice(1))
+}
+
+
+function collectStrings(obj) {
+    var newArr = [];
+
+    function collectStringsHelper(obj) {
+        for(var i=0; i<Object.keys(obj).length; i++) {
+            if(typeof obj[Object.keys(obj)[i]] == 'string')
+                newArr.push(obj[Object.keys(obj)[i]]);
+            else {
+                return collectStringsHelper(obj[Object.keys(obj)[i]]);
+            }
+        }
+    }
+
+    collectStringsHelper(obj)
+
+    return newArr;
+}
+
+
+function contains(obj, value) {
+    for(var key in obj) {
+        if(typeof obj[key] != 'object') {
+            if(obj[key] === value) {
+                return true;
+            }
+        } else {
+            contains(obj[key], value);
```

Comment:
Very close! You just need one more step over here to account for when you have multiple objects.